### PR TITLE
Rename Extension to SchemaExtension

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

To pave the way for FieldExtensions. Also add a deprecation warning when importing Extension.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Jonathan Kim for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by botberry** [comment]
## Apollo Federation Subgraph Compatibility Results

<table>
<thead>
<tr><th>Federation 1 Support</th><th>Federation 2 Support</th></tr>
</thead>
<tbody>
<tr><td><table><tr><th><code>_service</code></th><td>🟢</td></tr><tr><th><code>@key (single)</code></th><td>🟢</td></tr><tr><th><code>@key (multi)</code></th><td>🟢</td></tr><tr><th><code>@key (composite)</code></th><td>🟢</td></tr><tr><th><code>repeatable @key</code></th><td>🟢</td></tr><tr><th><code>@requires</code></th><td>🟢</td></tr><tr><th><code>@provides</code></th><td>🟢</td></tr><tr><th><code>federated tracing</code></th><td>🔲</td></tr></table></td><td><table><tr><th><code>@link</code></th><td>🟢</td></tr><tr><th><code>@shareable</code></th><td>🟢</td></tr><tr><th><code>@tag</code></th><td>🟢</td></tr><tr><th><code>@override</code></th><td>🟢</td></tr><tr><th><code>@inaccessible</code></th><td>🟢</td></tr><tr><th><code>@composeDirective</code></th><td>🟢</td></tr><tr><th><code>@interfaceObject</code></th><td>🟢</td></tr></table></td></tr>
</tbody>
</table>

Learn more:
* [Apollo Federation Subgraph Specification](https://www.apollographql.com/docs/federation/subgraph-spec/)
* [Compatibility Tests](https://github.com/apollographql/apollo-federation-subgraph-compatibility/blob/main/COMPATIBILITY.md)

File: tests/django/app/schema.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+@strawberry.type
+class Query:
+    hello: str = "🍓"
+
+    @strawberry.field
+    def hi(self, name: str) -> str:
+        return f"Hi {name}!"
+
+
+schema = strawberry.Schema(query=Query, mutation=Mutation, extensions=[MyExtension])
```
Comment (patrick91) [inline tests/django/app/schema.py:57]: is this file needed?

**Review by patrick91 — COMMENTED** [review]
looks great! Shall we add a line here? https://strawberry.rocks/docs/breaking-changes

even if it is not a breaking change? 🤔 (we could rename the page)

/cc @erikwrede @skilkis

File: tests/django/app/schema.py (hunk 1 above)
Comment (jkimbo) [reply tests/django/app/schema.py:57]: Hmm I think it got pulled in as part of a rebase. Will remove

**Comment by jkimbo** [comment]
@patrick91 

> looks great! Shall we add a line here? [strawberry.rocks/docs/breaking-changes](https://strawberry.rocks/docs/breaking-changes)

I can do but what version will this release be?

**Comment by patrick91** [comment]
> @patrick91
> 
> > looks great! Shall we add a line here? [strawberry.rocks/docs/breaking-changes](https://strawberry.rocks/docs/breaking-changes)
> 
> I can do but what version will this release be?

We can add/update the file before we merge the PR :D

File: strawberry/extensions/validation_cache.py [hunk 2]
```diff
@@ -31,6 +30,8 @@ class ValidationCache(Extension):
     """
 
     def __init__(self, maxsize: Optional[int] = None):
+        from strawberry.schema.execute import validate_document
+
         self.cached_validate_document = lru_cache(maxsize=maxsize)(validate_document)
```
Comment (erikwrede) [inline strawberry/extensions/validation_cache.py:35]: Is this change needed?

**Review by erikwrede — COMMENTED** [review]
Great change to streamline the naming for FieldExtensions!

File: strawberry/extensions/validation_cache.py (hunk 2 above)
Comment (jkimbo) [reply strawberry/extensions/validation_cache.py:35]: I don't think so. Will revert

**Review by erikwrede — APPROVED** [review]

**Review by patrick91 — APPROVED** [review]
Thank you!
